packages/client/lib/OpenID4VCIClient.ts

```diff
@@ -71,37 +71,55 @@ export class OpenID4VCIClient {
     return this._serverMetadata;
   }
 
-  public createAuthorizationRequestUrl({ clientId, codeChallengeMethod, codeChallenge, redirectUri, scope }: AuthorizationRequestOpts): string {
-    if (!scope) {
-      throw Error('Please provide a scope. authorization_details based requests are not supported at this time');
+  public createAuthorizationRequestUrl({ clientId, codeChallengeMethod, codeChallenge, authorizationDetails, redirectUri, scope }: AuthorizationRequestOpts): string {
+
+    // Scope and authorization_details can be used in the same authorization request
+    // https://datatracker.ietf.org/doc/html/draft-ietf-oauth-rar-23#name-relationship-to-scope-param
+    if (!scope && !authorizationDetails) {
+      throw Error('Please provide a scope or authorization_details');
     }
 
     if (!this._serverMetadata?.openid4vci_metadata?.authorization_endpoint) {
       throw Error('Server metadata does not contain authorization endpoint');
     }
 
     // add 'openid' scope if not present
-    if (!scope.includes('openid')) {
+    if (scope && !scope.includes('openid')) {
       scope = `openid ${scope}`;
     }
 
+    this.addLocations(authorizationDetails );
+
     const queryObj: AuthorizationRequest = {
       response_type: ResponseType.AUTH_CODE,
       client_id: clientId,
       code_challenge_method: codeChallengeMethod,
       code_challenge: codeChallenge,
+      authorization_details: JSON.stringify(authorizationDetails),
```

Comment:
Hmzz, this probably should be encoded right?

Reply:
It gets encoded at line 110

Reply:
That is for the whole object. Have you checked whether the invidivual param value should not already be encoded to begin with. Most oAuth2 and OpenID4SSI specs are mentioning that on specific params

Reply:
The `authorization_details` property is an object with 3 defined properties: `type` (mandatory), `format` (mandatory) and `locations` (optional). The examples provide additional properties, but they don't show up consistently among the examples and they don't show up anywhere else in the documentation. The examples below show the json string encoded:

[OpenID4VCI](https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#section-5.1.1-7)
```
GET /authorize?
  response_type=code
  &client_id=s6BhdRkqt3
  &code_challenge=E9Melhoa2OwvFrEMTJguCHaoeK1t8URWbuGJSstw-cM
  &code_challenge_method=S256
  &authorization_details=%5B%7B%22type%22:%22openid_credential
  %22,%22format%22:%22jwt_vc_json%22,%22types%22:%5B%22Verifia
  bleCredential%22,%22UniversityDegreeCredential%22%5D%7D%5D
  &redirect_uri=https%3A%2F%2Fclient.example.org%2Fcb
Host: https://server.example.com
``` 

[draft-ietf-oauth-rar-23 ](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-rar-23#name-authorization-request)
GET /authorize?response_type=code
   &client_id=s6BhdRkqt3
   &state=af0ifjsldkj
   &redirect_uri=https%3A%2F%2Fclient.example.org%2Fcb
   &code_challenge_method=S256
   &code_challenge=K2-ltc83acc4h0c9w6ESC_rEMTJ3bwc-uCHaoeK1t8U
   &authorization_details=%5B%7B%22type%22%3A%22account%5Finfo
   rmation%22%2C%22actions%22%3A%5B%22list%5Faccounts%22%2C%22
   read%5Fbalances%22%2C%22read%5Ftransactions%22%5D%2C%22loca
   tions%22%3A%5B%22https%3A%2F%2Fexample%2Ecom%2Faccounts%22%
   5D%7D%2C%7B%22type%22%3A%22payment%5Finitiation%22%2C%22act
   ions%22%3A%5B%22initiate%22%2C%22status%22%2C%22cancel%22%5
   D%2C%22locations%22%3A%5B%22https%3A%2F%2Fexample%2Ecom%2Fp
   ayments%22%5D%2C%22instructedAmount%22%3A%7B%22currency%22%
   3A%22EUR%22%2C%22amount%22%3A%22123%2E50%22%7D%2C%22credito
   rName%22%3A%22Merchant%20A%22%2C%22creditorAccount%22%3A%7B
   %22iban%22%3A%22DE02100100109307118603%22%7D%2C%22remittanc
   eInformationUnstructured%22%3A%22Ref%20Number%20Merchant%22
   %7D%5D HTTP/1.1
Host: server.example.com

Both show everything encoded